cmd/pktvisord/NetworkInterfaceScan.h

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#if __has_include(<ifaddrs.h>)
+#include <net/if.h>
+#include <ifaddrs.h>
+#else
+// System without ifaddrs
+#endif
+#include <string>
+
+namespace visor {
+
+static inline std::string most_used_interface()
+{
+#if __has_include(<ifaddrs.h>)
+    struct ifaddrs *ifaddr;
+    if (getifaddrs(&ifaddr) == -1) {
+        return std::string();
+    }
+    std::string interface;
+    uint64_t most_used = 0;
+    for (struct ifaddrs *ifa = ifaddr; ifa != nullptr; ifa = ifa->ifa_next) {
+        if (ifa->ifa_addr == nullptr || (ifa->ifa_flags & IFF_LOOPBACK) || !(ifa->ifa_flags & IFF_UP)) {
+            continue;
+        }
+#if defined(AF_PACKET)
+        if (int family = ifa->ifa_addr->sa_family; family == AF_PACKET && ifa->ifa_data != nullptr) {
+#elif defined(AF_LINK)
+        if (int family = ifa->ifa_addr->sa_family; family == AF_LINK && ifa->ifa_data != nullptr) {
+#else
+        if (false) {
+#endif
+            uint32_t *stats = reinterpret_cast<uint32_t *>(ifa->ifa_data);
```

Comment:
I think we should follow the man pages and use the actual stat struct rather than trying to interpret it directly like this. For example ```struct rtnl_link_stats *stats = ifa->ifa_data;``` from https://man7.org/linux/man-pages/man3/getifaddrs.3.html

The OSX version says ```     The ifa_data field references address family specific data.  For AF_LINK addresses it contains a pointer to the struct if_data (as defined in include file <net/if.h>) which contains various interface
     attributes and statistics.  For all other address families, it contains a pointer to the struct ifa_data (as defined in include file <net/if.h>) which contains per-address interface statistics.```

Reply:
I did that in that way because MacOS does not have `rtnl_link_stats` struct, and net/if does not have an interface to proper parser tx and rx packets, thats why I would like you to test this PR on MacOS so we know if it does work as expected or Not.

Reply:
Right but it contains a different structure which does have statistics. So that part would have to be in an #ifdef for each platform

Reply:
I've changed that by platform and added rtnl_link_stats for linux